src/coreclr/jit/codegencommon.cpp

```diff
@@ -1880,7 +1880,30 @@ void CodeGen::genGenerateMachineCode()
 
         if (compiler->fgHaveProfileData())
         {
-            printf("; with PGO: edge weights are %s, and fgCalledCount is " FMT_WT "\n",
+            const char* pgoKind;
+            switch (compiler->fgPgoSource)
+            {
+                case ICorJitInfo::PgoSource::Static:
+                    pgoKind = "Static";
+                    break;
+                case ICorJitInfo::PgoSource::Dynamic:
+                    pgoKind = "Dynamic";
+                    break;
+                case ICorJitInfo::PgoSource::Blend:
+                    pgoKind = "Blend";
+                    break;
+                case ICorJitInfo::PgoSource::Text:
+                    pgoKind = "Text";
+                    break;
+                case ICorJitInfo::PgoSource::Sampling:
+                    pgoKind = "Sampling";
+                    break;
```

Comment:
Just some nitpicking of the wording, feel free to ignore...
```suggestion
                case ICorJitInfo::PgoSource::Text:
                    pgoKind = "Textual";
                    break;
                case ICorJitInfo::PgoSource::Sampling:
                    pgoKind = "Sample-based";
                    break;
```